modules/priceFloors.js

```diff
@@ -372,6 +391,30 @@ function validateRules(floorsData, numFields, delimiter) {
   return Object.keys(floorsData.values).length > 0;
 }
 
+function modelIsValid(model) {
+  // schema.fields has only allowed attributes
+  if (!validateSchemaFields(utils.deepAccess(model, 'schema.fields'))) {
+    return false;
+  }
+  return validateRules(model, model.schema.fields.length, model.schema.delimiter || '|')
+}
+
+/**
+ * @summary Fields array should have at least one entry and all should match allowed fields
+ * Each rule in the values array should have a 'key' and 'floor' param
+ */
+const floorsSchemaValidation = {
+  1: data => modelIsValid(data),
+  2: data => {
+    // model groups should be an array with at least one element
+    if (!Array.isArray(data.modelGroups) || data.modelGroups.length === 0) {
+      return false;
+    }
+    // every model should have valid schema, as well as an accompanying modelWeight
+    return data.modelGroups.every(model => typeof model.modelWeight === 'number' && modelIsValid(model));
```

Comment:
Do all models need to be valid? Or could this work with whatever the list of valid models is?

Reply:
We could change this to a `filter`, instead of an `every`. I'll ask Bryan what he thinks.

Would want to add some error logging for a model which does not fit though.

Reply:
Hey @msm0504 , while this is a good idea in spirit, in practice there would be some major implications if a floor provider is expecting to have a certain weight, then all the sudden it's increased due to a model not applied. This would be a difficult conversation with the publisher if a floor provider was reducing a model that was monetizing poorly and al the sudden it shot up to 100% (in an extreme example) if the new model had an error.

Having a binary working / not working system would alert the floor provider if an issue exists, where no floors would be used in this case. Having a silent issue could be more costly.

I appreciate thorough review Mark. Feedback is always welcome.

Bryan